src/element/dragElements.ts

```diff
@@ -6,44 +6,75 @@ import { getPerfectElementSize } from "./sizeHelpers";
 import Scene from "../scene/Scene";
 import { NonDeletedExcalidrawElement } from "./types";
 import { PointerDownState } from "../types";
+import { getBoundTextElementId } from "./textElement";
 
 export const dragSelectedElements = (
   pointerDownState: PointerDownState,
   selectedElements: NonDeletedExcalidrawElement[],
   pointerX: number,
   pointerY: number,
-  scene: Scene,
   lockDirection: boolean = false,
   distanceX: number = 0,
   distanceY: number = 0,
 ) => {
   const [x1, y1] = getCommonBounds(selectedElements);
   const offset = { x: pointerX - x1, y: pointerY - y1 };
   selectedElements.forEach((element) => {
-    let x: number;
-    let y: number;
-    if (lockDirection) {
-      const lockX = lockDirection && distanceX < distanceY;
-      const lockY = lockDirection && distanceX > distanceY;
-      const original = pointerDownState.originalElements.get(element.id);
-      x = lockX && original ? original.x : element.x + offset.x;
-      y = lockY && original ? original.y : element.y + offset.y;
-    } else {
-      x = element.x + offset.x;
-      y = element.y + offset.y;
+    updateElementCoords(
+      lockDirection,
+      distanceX,
+      distanceY,
+      pointerDownState,
+      element,
+      offset,
+    );
+    if (!element.groupIds.length) {
```

Comment:
this check means that any text-bound container will not behave atomically when it's also part of a group (and you're editing the group)

Reply:
https://github.com/excalidraw/excalidraw/issues/4456